src/shared/libs/iota/quorum.js

```diff
@@ -0,0 +1,415 @@
+ *
+ *   @method findSyncedNodes
+ *   @param {array} nodes
+ *   @param {array} [syncedNodes = []]
+ *   @param {array} [blacklistedNodes = []]
+ *
+ *   @returns {Promise}
+ **/
+const findSyncedNodes = (nodes, quorumSize, syncedNodes = [], blacklistedNodes = []) => {
+    const sizeOfSyncedNodes = size(syncedNodes);
+    const whitelistedNodes = filter(nodes, (node) => !includes(blacklistedNodes, node) && !includes(syncedNodes, node));
+
+    if (isEmpty(whitelistedNodes)) {
+        return Promise.reject(new Error(Errors.NOT_ENOUGH_HEALTHY_QUORUM_NODES));
+    }
+
+    const selectedNodes =
+        sizeOfSyncedNodes === quorumSize ? syncedNodes : sampleSize(whitelistedNodes, quorumSize - sizeOfSyncedNodes);
+
+    return Promise.all(map(selectedNodes, (provider) => isNodeSynced(provider).catch(() => undefined))).then(
+        (results) => {
+            const { activeNodes, inactiveNodes } = transform(
+                selectedNodes,
+                (acc, node, idx) => (results[idx] ? acc.activeNodes.push(node) : acc.inactiveNodes.push(node)),
+                { activeNodes: [], inactiveNodes: [] },
+            );
+
+            if (size(activeNodes) === size(selectedNodes)) {
+                return [...syncedNodes, ...activeNodes];
+            }
+
+            return findSyncedNodes(
+                nodes,
+                quorumSize,
+                // Add active nodes to synced nodes
+                [...syncedNodes, ...activeNodes],
+                // Add inactive nodes to blacklisted nodes
+                [...blacklistedNodes, ...inactiveNodes],
+            );
+        },
+    );
+};
+
+/**
+ *   From a list of synced nodes, compute a quorum result for wereAddressesSpentFrom iota api
+ *
+ *   @method getQuorumForWereAddressesSpentFrom
+ *   @param {array} payload - Addresses
+ *   @param {array} syncedNodes
+ *
+ *   @returns {Promise}
+ **/
+const getQuorumForWereAddressesSpentFrom = (payload, syncedNodes) => {
+    const requestPayloadSize = size(payload);
+
+    return rejectIfNotEnoughSyncedNodes(syncedNodes)
+        .then(() =>
+            Promise.all(
+                map(
+                    syncedNodes,
+                    (provider) =>
+                        new Promise((resolve) => {
+                            new IOTA({ provider }).api.wereAddressesSpentFrom(
+                                payload,
+                                (err, spendStatuses) => (err ? resolve(undefined) : resolve(spendStatuses)),
+                            );
+                        }),
+                ),
+            ),
+        )
+        .then((results) => {
+            const validResults = filter(results, (result) => !isUndefined(result));
+
+            let idx = 0;
+            const quorumResult = [];
+
+            while (idx < requestPayloadSize) {
+                /* eslint-disable no-loop-func */
+                quorumResult.push(
```

Comment:
IIUC, if the request is for L addresses the Quorom is determined for each address separately, correct?
if so, that's great! (what I envisioned) but it should be clearly stated (in the JSDocs?)

Reply:
That's correct. Quorum is determined for each address separately. 
Have added better code comments and also have updated [JSDocs](https://github.com/laumair/trinity-wallet/blob/feat/quorum/src/shared/libs/iota/quorum.js#L167-L171).